src/fineman/elimination_algorithm.py

```diff
@@ -0,0 +1,91 @@
+from math import ceil
+
+from fineman.betweenness_reduction import betweenness_reduction
+from fineman.elimination_by_hop_reduction import elimination_of_r_remote_edges_by_hop_reduction
+from fineman.helper_functions import reweight_graph, super_source_bfd, compute_reach, transpose_graph, b_hop_sssp, \
+    b_hop_stsp
+from fineman.independent_set_or_crust import find_is_or_crust
+
+
+def _eliminate_independent_set(graph, independent_set):
+    out_I = {(u,v) for u in independent_set for v in graph[u].keys()}
+    graph_out_I, _ = _subgraph_with_out_set(graph, out_I)
+    return super_source_bfd(graph_out_I, out_I, 1)
+
+
+def _compute_price_functiom_to_make_U_r_remote(graph, neg_edges, negative_sandwich, beta):
+
+    (x,U,y) = negative_sandwich
+    phi = [] * len(graph)
+
+    dists_from_x= b_hop_sssp(x, graph, neg_edges, beta)
+    dists_to_y = b_hop_stsp(y, graph, beta)
+
+    for v in graph.keys():
+        phi[v] = min(0, max(dists_from_x[v], -(dists_to_y[v])))
+
+    return phi
+
+
+def _subgraph_with_out_set(graph: dict[int, dict[int, int]], out_set: set):
+    new_graph = {}
+    new_neg_edges = set()
+
+    for u, edge in graph.items():
+        if u not in new_graph:
+            new_graph[u] = {}
+        for v, w in edge.items():
+            if v not in out_set:
+                new_graph[u][v] = w
+                if w < 0:
+                    new_neg_edges.add((u,v))
+
+    return new_graph, new_neg_edges
+
+
+def elimination_algorithm(org_graph, org_neg_edges):
+
+    n = len(org_graph.keys())
+
+    # TODO: probably not efficient to compute these each time.
+    neg_vertices = {u for u,_ in org_neg_edges}
+
+    c = 3
+    k = len(org_neg_edges)
+    r = ceil(k**(1/9))
+
+
+    phi_1 = betweenness_reduction(org_graph, org_neg_edges, tau=r, beta=r+1, c=c)
```

Comment:
We should make a decision on whether the functions that compute price functions apply them, themselves or if they return them and we apply them in the main function. This should be consistent throughout the codebase.